internal/checker/cloudformation/rules/encryption/kms_key_policy_no_wildcard.rego

```diff
@@ -0,0 +1,66 @@
+# METADATA
+# title: "Verify that the KMS key policy does not incorporate any wildcard (*) principals."
+# description: "A wildcard principal serves as a placeholder granting access to all users or accounts, posing a risk of unauthorized access to your KMS keys. Eliminating wildcard principals from your key policies ensures that only designated users or accounts can access your KMS keys, enhancing key security and mitigating the risk of unauthorized entry."
+# scope: package
+# related_resources:
+# - https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-kms-key.html
+# custom:
+#   id: CB_CFT_27
+#   severity: HIGH
+package lib.cloudformation.CB_CFT_27
+import future.keywords.in
+
+resource := "AWS::KMS::Key"
+
+is_valid {
+	some resources in input.Resources
+	resources.Type == resource
+}
+
+contains_wildcard_principal(statements) {
+	some statement in statements
+	statement.Principal == "*"
+}
+
+aws_principal(statements) {
+	some statement in statements
+	statement.Principal.AWS == "*"
+}
+
+contains_deny_effect(statements) {
+	some statement in statements
+	statement.Effect == "Deny"
+}
+
```

Comment:
swap the logic for pass and fail so that it will only fail if conditions are met
fail[properties] {
	is_valid
	some resources in input.Resources
	properties := resources.Properties.KeyPolicy
	statements := properties.Statement
	aws_principal(statements)
	contains_wildcard_principal(statements)
	contains_deny_effect(statements)
}

pass[resources] {
	resources := input.Resources
	is_valid
	count(fail) == 0
}